pinot-core/src/main/java/org/apache/pinot/core/operator/combine/GroupByOrderByCombineOperator.java

```diff
@@ -125,43 +119,41 @@ public String getOperatorName() {
    */
   @Override
   protected void processSegments(int threadIndex) {
-    try {
-      IntermediateResultsBlock intermediateResultsBlock =
-          (IntermediateResultsBlock) _operators.get(threadIndex).nextBlock();
-
-      _initLock.lock();
-      try {
-        if (_dataSchema == null) {
-          _dataSchema = intermediateResultsBlock.getDataSchema();
-          // NOTE: Use trimSize as resultSize on server size.
-          if (_trimThreshold >= MAX_TRIM_THRESHOLD) {
-            // special case of trim threshold where it is set to max value.
-            // there won't be any trimming during upsert in this case.
-            // thus we can avoid the overhead of read-lock and write-lock
-            // in the upsert method.
-            _indexedTable = new UnboundedConcurrentIndexedTable(_dataSchema, _queryContext, _trimSize);
-          } else {
-            _indexedTable =
-                new ConcurrentIndexedTable(_dataSchema, _queryContext, _trimSize, _trimSize, _trimThreshold);
+    for (int operatorIndex = threadIndex; operatorIndex < _numOperators; operatorIndex += _numThreads) {
```

Comment:
in group by case, _numOperators will continue to be equal to _numThreads?

Reply:
So when you say `This PR doesn't limit the thread usage for group-by combine operator to keep the existing behavior, but only make it possible` you mean that we now have option to pass a different numThreads to this method? And that was completely impossible previously?

Reply:
The existing code in the group-by combine operator has the assumption that one thread only processes one segment. This PR removes that assumption, and each thread can process multiple segments. Currently we still use `_numOperators` threads to process the query, but if we give less threads, it still works